### DISCUSSION TITLE
Publish and Refresh Direct Lake models

### DISCUSSION THREAD

**Discussion by pawarbi** [TabularEditor/TabularEditor3#1230] [Ideas]
### Is your feature request related to a problem? Please describe

I updated [my blog post](https://fabric.guru/controlling-direct-lake-fallback-behavior#:~:text=Semantic%20model%20is%20published%20via%20XMLA%20endpoint%20and%20has%20not%20been%20reframed.) yesterday about publishing a DL model via TE3. It works great but the current DL limitation is that it needs to be refreshed after publishing. Most users wont realize it, in fact in your own [video/blog](https://blog.tabulareditor.com/2023/09/26/fabric-direct-lake-with-tabular-editor-part-2-creation/), you missed that step.

### Describe the solution you'd like

The solution is easy - just process the tables. But I was wondering if you can do it automatically if the Direct Lake connector is used. Or give users a check box and all DL models should be refreshed after publishing. It's a metadata refresh so wont take longer or affect the capacity. 

### Goals

1.
2.

**Comment by otykier** [comment]
Yeah, we were recently made aware about this issue. I think we’ll work towards a solution where TE automatically performs the refresh upon the first deployment of a DL model. That should do the trick.

**Comment by mlonsk** [comment]
This was implemented in Tabular Editor 3.15.0 and later versions.

> When a Direct Lake model is initially deployed, we now perform an automatic refresh, which is a prerequisite to avoid DirectQuery fallback. Moreover, there is a new option (which is enabled by default), to also perform an automatic refresh whenever a Direct Lake model is saved, when new tables have been added. This preference is located under Tools > Preferences > Model Deployment > Data Refresh.

Sorry for not getting this updated sooner. Closing the discussion
